dbms/src/Storages/DeltaMerge/DeltaMergeStore.cpp

```diff
@@ -1485,7 +1485,13 @@ void DeltaMergeStore::checkSegmentUpdate(const DMContextPtr & dm_context, const
         if (shutdown_called.load(std::memory_order_relaxed))
             return;
 
-        auto [added, heavy] = background_tasks.tryAddTask(task, thread_type, std::max(id_to_segment.size() * 2, background_pool.getNumberOfThreads() * 3), log);
+        size_t segment_size;
+        {
+            std::shared_lock lock(read_write_mutex);
+            segment_size = id_to_segment.size();
```

Comment:
Acquire a lock on this task seems too heavy. Can we use a relaxed atomic variable here to record the number of segments?

Reply:
I am thinking about whether it is necessary to get an accurate `segment_size`.

Reply:
The mutex is used in write path which is a hot path while there are many background tasks under heavy write scenario. And the accurate value of segment size is not important here. So better to avoid the unnecessary cost here.

Reply:
Seems that this lock may be blocked for long time by other heavy segment mutation tasks, which involves IO. I will change to use something else.